lib/presentation/shared_widgets/commitments/commitment_card_list.dart

```diff
@@ -65,12 +65,15 @@ class CommitmentCardListState extends State<CommitmentCardList> {
     setState(() {
       // Remove from active commitments
       _activeCommitments.remove(option.id);
-      // Check if has required
-      final children = option.requires;
-      // If yes - loop & deselect all children
-      if (children != null) {
-        _deselectAll(children);
+
+      // Recursively deselect all parents
+      for (final commitment in _commitments
```

Comment:
I think we should abstract the `.where` to a new variable for readability sake, the variable name can define what is happening and make it easier to understand what `List` we are iterating.